src/main/kotlin/pl/edu/uj/ii/ksi/mordor/persistence/managers/MetadataManager.kt

```diff
@@ -0,0 +1,21 @@
+package pl.edu.uj.ii.ksi.mordor.persistence.managers
+
+import java.io.File
+import pl.edu.uj.ii.ksi.mordor.persistence.entities.FileContent
+import pl.edu.uj.ii.ksi.mordor.persistence.entities.FileEntry
+import pl.edu.uj.ii.ksi.mordor.persistence.entities.FileMetadata
+import pl.edu.uj.ii.ksi.mordor.persistence.entities.FileThumbnail
+
+interface MetadataManager {
+    fun extractAndSaveMetadata(file: File): FileMetadata
+
+    fun getAllEntries(metadata: FileMetadata): Set<FileEntry>
```

Comment:
If you return Array instead of Set, client will not be forced to take care of ordering the results.

Reply:
Array does not provide unique results. 

Why should the results be ordered?

Reply:
Because client will display these entries. 
Taking care of ordering them by default would be easy to achieve by this manager, because of how the database works.

 
Take a look at this: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/distinct-by.html